agent/log_streamer.go

```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/buildkite/agent/v3/status"
 )
 
+const jobLogMaxSize = 1024 * 1024 * 1024 // 1 GiB; fits in int32
```

Comment:
I think the server will need to tell the agent this number per-job, probably in the response to AcceptJob.
The limit might change over time, or vary by plan, etc.

Reply:
Makes sense. I've added a field to `api.Job` and turned this constant into a default.